packages/core/src/rules/oas3/spec-ref-validation.ts

```diff
@@ -0,0 +1,50 @@
+import { Oas3Rule, VisitFunction } from '../../visitors';
+import { isRef } from '../../ref-utils';
+import { UserContext } from '../../walk';
+
+export const SpecRefValidation: Oas3Rule = () => {
+  function validateNoRef(node: unknown, { report, rawLocation }: UserContext) {
+    if (isRef(node)) {
+      report({
+        message: 'Invalid usage of $ref',
+        location: rawLocation.child('$ref').key(),
+      });
+    }
+  }
```

Comment:
Do we need a separate function? Why don't put it simply in place [here](https://github.com/Redocly/redocly-cli/pull/1112/files?diff=unified&w=1#diff-7a126114b978592806f48c748aecefa06259fc940e41d8f3dcdc410db5380b44R44)?